hummingbot/market/bittrex/bittrex_api_order_book_data_source.py

```diff
@@ -0,0 +1,394 @@
+        raise IOError
+
+    async def get_tracking_pairs(self) -> Dict[str, OrderBookTrackerEntry]:
+        # Get the current active markets
+        trading_pairs: List[str] = await self.get_trading_pairs()
+        retval: Dict[str, OrderBookTrackerEntry] = {}
+
+        number_of_pairs: int = len(trading_pairs)
+        for index, trading_pair in enumerate(trading_pairs):
+
+            # TODO: Refactor accordingly when V3 WebSocket API is released
+            # get_snapshot() utilizes WebSocket API. Requires market symbols in 'Quote-Base' format
+            # Code below converts 'Base-Quote' -> 'Quote-Base'
+            temp_trading_pair = f"{trading_pair.split('-')[1]}-{trading_pair.split('-')[0]}"
+
+            try:
+                snapshot: Dict[str, any] = await self.get_snapshot(temp_trading_pair)
+                snapshot_timestamp: float = snapshot["nonce"]
+                snapshot_msg: OrderBookMessage = self.order_book_class.snapshot_message_from_exchange(
+                    snapshot["results"], snapshot_timestamp, metadata={"symbol": trading_pair}
+                )
+                order_book: BittrexOrderBook = BittrexOrderBook()
+                active_order_tracker: BittrexActiveOrderTracker = BittrexActiveOrderTracker()
+
+                bids, asks = active_order_tracker.convert_snapshot_message_to_order_book_row(snapshot_msg)
+                order_book.apply_snapshot(bids, asks, snapshot_msg.update_id)
+                retval[trading_pair] = BittrexOrderBookTrackerEntry(
+                    trading_pair, snapshot_timestamp, order_book, active_order_tracker
+                )
+                self.logger().info(
+                    f"Initialized order book for {trading_pair}. " f"{index + 1}/{number_of_pairs} completed."
+                )
+                await asyncio.sleep(0.6)
+            except IOError:
+                self.logger().network(
+                    f"Max retries met fetching snapshot for {trading_pair}.",
+                    exc_info=True,
+                    app_warning_msg=f"Error getting snapshot fro {trading_pair}. Check network connection.",
+                )
+            except Exception:
+                self.logger().error(f"Error initiailizing order book for {trading_pair}. ", exc_info=True)
+                await asyncio.sleep(5.0)
+        return retval
+
+    async def listen_for_trades(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+        # Trade messages are received from the order book web socket
+        pass
+
+    async def _socket_stream(self, conn: signalr_aio.Connection) -> AsyncIterable[str]:
+        try:
+            while True:
+                async with timeout(MESSAGE_TIMEOUT):
+                    yield await conn.msg_queue.get()
+        except asyncio.TimeoutError:
+            self.logger().warning("Message recv() timed out. Going to reconnect...")
+            return
+
+    async def _transform_raw_message(self, msg) -> Dict[str, Any]:
```

Comment:
This function is repeated a few times in the data source code. You should refactor it out.